src/lib/components/ui/ScrollView/ScrollView.jsx

```diff
@@ -75,6 +77,22 @@ export const ScrollView = (props) => {
     debounce,
   );
 
+  const autoScrollChildrenHash = autoScroll && children && React.Children
+    .map(children, (child) => child.key)
+    .reduce((reducedHash, childHash) => reducedHash + childHash, '');
+  const autoScrollChildrenLength = autoScroll && children && children.length;
+
+  useLayoutEffect(() => {
```

Comment:
This is not working for me reliably. I tested it using the documentation demo and after a few clicks the component did not scroll to the end. I was expecting to always see the last added text and that is not happening.

Attached is a screenshot of state that I repeatably achieved in both Chrome and Firefox (on Linux) by clicking the `Add text into scroll view` only.

![Screenshot from 2020-07-07 15-45-29](https://user-images.githubusercontent.com/4589176/86790625-ed625e00-c068-11ea-96dc-ef4c7317de99.png)

Reply:
I confirm this behaviour. The vertical version is fine but the horizontal one is having problems a few clicks (Chrome/macOS).

Reply:
I noticed the same problems for both vertical and horizontal.

Reply:
This was partly bug in existing `getElementsPositionDifference` function.  Internally, it use `getBoundingClientRect()` that sometimes returns floating point value that result in incorrect detection of the end of scrollable area that results in not-working autoscroll.

Next problem is that if animation is not completed and new content is added than end of scrollable area is not detected so it do not autoscroll. I will fix this.